mockgcp/mockgkemulticloud/attachedcluster.go

```diff
@@ -66,6 +67,28 @@ func (s *GKEMulticloudV1) CreateAttachedCluster(ctx context.Context, req *pb.Cre
 	obj := proto.Clone(req.AttachedCluster).(*pb.AttachedCluster)
 	obj.Name = fqn
 
+	if obj.GetBinaryAuthorization() == nil {
+		obj.BinaryAuthorization = &pb.BinaryAuthorization{
+			EvaluationMode: pb.BinaryAuthorization_DISABLED,
+		}
+	}
+	if obj.GetMonitoringConfig() == nil {
+		obj.MonitoringConfig = &pb.MonitoringConfig{
+			ManagedPrometheusConfig: &pb.ManagedPrometheusConfig{},
+		}
+	}
+	obj.Fleet.Membership = obj.Fleet.Project + "/locations/global/memberships/" + req.AttachedClusterId
+	obj.CreateTime = timestamppb.New(now)
+	obj.UpdateTime = timestamppb.New(now)
+	obj.Etag = "abcdef0123A="
+	ver, err := trimPlatformVersion(req.GetAttachedCluster().GetPlatformVersion())
```

Comment:
Is `spec.platformVersion` required? My gut feeling is that the platform version is associated with the fleet membership which means it could have a default value if unset.

Reply:
It is a required value, no defaults. There's no correlation between the fleet membership and the platform version.

Reply:
Note that platform version is different from k8s version. K8s version is determined by the service by querying the underlying cluster. Platform version is the version of attached clusters that's being provisioned.

Reply:
Good to know. Thank you!